### PR TITLE
add some docs for AtlasDbImmutableTable

### PR DESCRIPTION
**Pull request by sixinli**
**Goals (and why)**:
trying to explain when to use which `getRowsColumnRange`

**Implementation Description (bullets)**:
- add docs for AtlasDbImmutableTable

**Concerns (what feedback would you like?)**:
- is the doc clear?

**Priority (whenever / two weeks / yesterday)**:
whenever

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2445)

### CODE REVIEW

File: atlasdb-client/src/main/java/com/palantir/atlasdb/table/api/AtlasDbImmutableTable.java [hunk 1]
```diff
@@ -36,9 +36,21 @@
...
+    
+    /*
+     * This returns visitables for each row. It should be used if you want to visit a number of the first 
+     * matched columns instead of all the columns per row.
+     */ 
     Map<ROW, BatchingVisitable<COLUMN_VALUE>> getRowsColumnRange(Iterable<ROW> rows,
                                                                  BatchColumnRangeSelection columnRangeSelection);
 
+    /*
+     * This returns an iteractor that visits the result columns row by row.
```
Comment (gmaretic) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/table/api/AtlasDbImmutableTable.java]: iterator

File: atlasdb-client/src/main/java/com/palantir/atlasdb/table/api/AtlasDbImmutableTable.java [hunk 2]
```diff
@@ -36,9 +36,21 @@
     Multimap<ROW, COLUMN_VALUE> getRowsMultimap(Iterable<ROW> rows);
     Multimap<ROW, COLUMN_VALUE> getRowsMultimap(Iterable<ROW> rows,
                                                 ColumnSelection columnSelection);
+    
+    /*
+     * This returns visitables for each row. It should be used if you want to visit a number of the first 
+     * matched columns instead of all the columns per row.
+     */
```
Comment (gmaretic) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/table/api/AtlasDbImmutableTable.java]: Either include the args and the return value for a javadoc or use regular comments.

File: atlasdb-client/src/main/java/com/palantir/atlasdb/table/api/AtlasDbImmutableTable.java [hunk 3]
```diff
@@ -36,9 +36,21 @@
...
     Map<ROW, BatchingVisitable<COLUMN_VALUE>> getRowsColumnRange(Iterable<ROW> rows,
                                                                  BatchColumnRangeSelection columnRangeSelection);
 
+    /*
+     * This returns an iteractor that visits the result columns row by row.
+     * 
+     * @param rows rows to get
+     * @param columnRangeSelection column range selection for each row
+     * @param batchHint batch size for reading from the database
+     */
```
Comment (gmaretic) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/table/api/AtlasDbImmutableTable.java]: Include return value

**Comment by sixinli** [comment]
oops know a way to get by the code coverage? it might not be happy because there are more lines :(

**Comment by gmaretic** [comment]
That check is not necessary for merging!

**Pull request merged**